src/metabase/util/date_2/parse.clj

```diff
@@ -146,5 +149,5 @@
 (defn parse
   "Parse a string into a `java.time` object."
   [^String s]
-  (or (try-parse-as-iso-timestamp s) ;; Try the fastpath first.
+  (or (and s (try-parse-as-iso-timestamp s)) ;; Try the fastpath first.
```

Comment:
nit: `try-` makes the function sound quite safe, but now it can NPE. I'd prefer leaving this as a `when` inside the function.

Reply:
Fair enough, updated.